test/TestUtils.ts

```diff
@@ -0,0 +1,426 @@
+
+  const { relayManagerAddress } = await waitForFunction(
+    getServerStatus,
+    url,
+    "can't ping server",
+    3
+  );
+
+  console.log('Relay Server Manager Address', relayManagerAddress);
+
+  await provider.getSigner(relayOwner).sendTransaction({
+    to: relayManagerAddress,
+    value: utils.parseEther('2'),
+  });
+
+  const relayHub = RelayHub__factory.connect(relayHubAddress, provider);
+
+  await relayHub.stakeForAddress(relayManagerAddress, delay || 2000, {
+    from: relayOwner,
+    value: stake || utils.parseEther('1'),
+  });
+
+  const { relayWorkerAddress } = await waitForFunction(
+    getServerReady,
+    url,
+    'timed out waiting for relay to get staked and registered',
+    25,
+    500
+  );
+
+  return {
+    proc,
+    worker: relayWorkerAddress,
+    manager: relayManagerAddress,
+  };
+};
+
+const buildServerUrl = () => {
+  const app = config.get<AppConfig>('app');
+
+  const portFromUrl = app.url.match(/:(\d{0,5})$/);
+
+  return !portFromUrl && app.port ? `${app.url}:${app.port}` : app.url;
+};
+
+const getServerStatus = async (
+  url: string,
+  client = new HttpClient(new HttpWrapper(undefined, 'silent'))
+): Promise<HubInfo> => client.getChainInfo(url);
+
+const getServerReady = async (url: string): Promise<HubInfo | undefined> => {
+  const response = await getServerStatus(url);
+
+  if (response.ready) {
+    return response;
+  }
+
+  return undefined;
+};
+
+const waitForFunction = async (
+  functionToWait: (url: string) => Promise<HubInfo | undefined>,
+  url: string,
+  errorMessage: string,
+  attempts: number,
+  interval = 1000
+): Promise<HubInfo> => {
+  for (let i = 0; i < attempts; i++) {
+    try {
+      console.log('sleep before cont.');
+      await sleep(interval);
+      const response = await functionToWait(url);
+
+      if (!response) {
+        continue;
+      }
+
+      return response;
+    } catch (e) {
+      console.log('getChainInfo failed');
+    }
+  }
+
+  throw Error(errorMessage);
+};
+
+const stopRelay = (proc: ChildProcessWithoutNullStreams): void => {
+  proc.kill();
+};
+
+const evmMine = async (): Promise<unknown> => {
+  return provider.send('evm_mine', []);
+};
+
+const evmMineMany = async (count: number): Promise<void> => {
+  for (let i = 0; i < count; i++) {
+    await evmMine();
+  }
+};
+
+const increaseBlockchainTime = async (time: number): Promise<void> => {
+  await provider.send('evm_increaseTime', [time]);
+  await evmMine();
+};
+
+const createSnapshot = async (): Promise<string> => {
+  return (await provider.send('evm_snapshot', [])) as string;
+};
+
+const revertSnapshot = async (snapshotId: string) => {
+  return (await provider.send('evm_revert', [snapshotId])) as boolean;
+};
+
+// An existing account in RSKJ that have been depleted
+const getExistingGaslessAccount = async (): Promise<Wallet> => {
+  const gaslessAccount = new Wallet(
+    '0x082f57b8084286a079aeb9f2d0e17e565ced44a2cb9ce4844e6d4b9d89f3f595',
```

Comment:
how do we know that this address will always work?

Reply:
actually Im assuming that is hardcoded in the node, I just followed what they had before, but I could change the approach to use the getAccount methods from the provider, but Im not pretty sure if its going to change something. @antomor what do you think?

Reply:
we need to discuss it

Reply:
It's the private key corresponding to one of the preloaded accounts. You can see it [here](https://github.com/rsksmart/rif-relay/blob/0e9c2d506483096e8549ed125145720b7ec6a8ad/test/EnvelopingUtils.test.ts#L183) and it corresponds to account at index 9 of the accounts retrieved with eth_accounts from regtest. I think we should: 
1. store it in a constant since it's used elsewhere
2. add a comment to explicitly describe where it comes from.